pkg/apis/ceph.rook.io/v1/types.go

```diff
@@ -964,9 +964,15 @@ type FilesystemSpec struct {
 	// +nullable
 	MetadataPool PoolSpec `json:"metadataPool"`
 
-	// The data pool settings
-	// +nullable
-	DataPools []PoolSpec `json:"dataPools"`
+	// The data pool settings, deprecated.
+	// At least DataPools or DataPoolsMapped should be specified
+	// +optional
+	DataPools []PoolSpec `json:"dataPools,omitempty"`
+
+	// The data pool settings with explicit data pool name.
+	// At least DataPools or DataPoolsMapped should be specified
```

Comment:
I assume it is not supported for users to specify both of these?

```suggestion
	// The data pool settings with explicit data pool name.
	// Exactly one of DataPools or DataPoolsMapped (preferred) must be specified.
```